src/middlewared/middlewared/api/v25_10_0/pool_snapshot.py

```diff
@@ -194,3 +195,21 @@ class PoolSnapshotUpdateArgs(BaseModel):
 
 class PoolSnapshotUpdateResult(BaseModel):
     result: PoolSnapshotCreateUpdateEntry
+
+
+class PoolSnapshotRenameOptions(BaseModel):
+    new_name: NonEmptyString
```

Comment:
Can we also validate snapshot names?

Reply:
Can we run `validate_snapshot_name` here? That'll be more convenient to use than to copypaste this validation from plugin to plugin.